### PR TITLE
Avoid creating a new EmojiMap for each post

### PR DESCRIPTION
**Pull request by hmhealey**
This change was suggested by @sudheerDev, but it turned out it was already part of one of my other PRs so we broke it out on it's own.

@esethna I'm proposing this for 4.5 since it leads to a fair bit less memory usage on pre-release, and the change is minimal since it works the same as before except it doesn't create as many copies of some stuff in memory.

### CODE REVIEW

**Comment by sudheerDev** [comment]
Always a pleasure.

File: selectors/emojis.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import {createSelector} from 'reselect';
+
+import {getCustomEmojisByName} from 'mattermost-redux/selectors/entities/emojis';
+
+import {EmojiMap} from 'stores/emoji_store.jsx';
+
+export const getEmojiMap = createSelector(
+    getCustomEmojisByName,
+    (customEmojisByName) => {
+        console.log('recalculating emojis1');
```
Comment (sudheerDev) [inline selectors/emojis.js]: Why the log?

Comment (hmhealey) [reply selectors/emojis.js]: Oops, that was for testing. That's also concerning that that wasn't caught by the style checker

Comment (sudheerDev) [reply selectors/emojis.js]: @hmhealey tell me about it!

**Comment by esethna** [comment]
Thanks @hmhealey @sudheerDev sounds good, we can take it

**Pull request merged**